dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3671,25 +3682,92 @@ Sedp::TypeLookupRequestReader::process_get_types_request(const XTypes::TypeLooku
-              ACE_TEXT("failed to deserialize type lookup reply\n")));
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: Sedp::TypeLookupReplyReader::process_type_lookup_reply - ")
+               ACE_TEXT("failed to deserialize type lookup reply\n")));
     return DDS::RETCODE_ERROR;
   }
 
-  if (type_lookup_reply.data.getTypes.result.types.length() > 0) {
-    sedp_.type_lookup_service_->add_type_objects_to_cache(type_lookup_reply.data.getTypes.result.types);
+  DDS::ReturnCode_t retcode;
+  switch (type_lookup_reply.data.kind) {
+  case XTypes::TypeLookup_getTypes_HashId:
+    retcode = process_get_types_reply(type_lookup_reply);
+    break;
+  case XTypes::TypeLookup_getDependencies_HashId:
+    retcode = process_get_dependencies_reply(sample, type_lookup_reply);
+    break;
+  default:
+    return DDS::RETCODE_UNSUPPORTED;
+  }
+
+  if (DDS::RETCODE_OK == retcode) {
     DCPS::SequenceNumber rpc_sequence;
     rpc_sequence.setValue(type_lookup_reply.header.related_request_id.sequence_number.high,
-      type_lookup_reply.header.related_request_id.sequence_number.low);
+                          type_lookup_reply.header.related_request_id.sequence_number.low);
+    // TODO(sonndinh): Is this a correct place to call match_continue?
     sedp_.match_continue(rpc_sequence);
   }
   return DDS::RETCODE_OK;
 }
 
+DDS::ReturnCode_t
+Sedp::TypeLookupReplyReader::process_get_types_reply(const XTypes::TypeLookup_Reply& reply)
+{
+  if (reply.data.getTypes.result.types.length() > 0) {
+    sedp_.type_lookup_service_->add_type_objects_to_cache(reply.data.getTypes.result.types);
+    return DDS::RETCODE_OK;
+  }
+  return DDS::RETCODE_NO_DATA;
+}
+
+DDS::ReturnCode_t
+Sedp::TypeLookupReplyReader::process_get_dependencies_reply(const DCPS::ReceivedDataSample& sample,
+                                                            const XTypes::TypeLookup_Reply& reply)
+{
+  const XTypes::TypeLookup_getTypeDependencies_Out& data = reply.data.getTypeDependencies.result;
+  continuation_point_ = data.continuation_point;
+  if (data.dependent_typeids.length() > 0) {
+    XTypes::TypeIdentifierSeq req_type_ids;
+    for (size_t i = 0; i < data.dependent_typeids.length(); ++i) {
+      const XTypes::TypeIdentifier& ti = data.dependent_typeids[i].type_id;
+      if (!sedp_.type_lookup_service_->type_object_in_cache(ti)) {
+        req_type_ids.append(ti);
+      }
+    }
+
+    DCPS::RepoId remote_guid = sample.header_.publication_id_;
+    if (!sedp_.send_type_lookup_request(req_type_ids, remote_guid, XTypes::TypeLookup_getTypes_HashId)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: Sedp::TypeLookupReplyReader::process_get_dependencies_reply - ")
+                 ACE_TEXT("failed to send type lookup request\n")));
+      return DDS::RETCODE_ERROR;
+    }
+    sedp_.type_lookup_reply_deadline_processor_->schedule(sedp_.max_type_lookup_service_reply_period_);
```

Comment:
`sedp_.type_lookup_reply_deadline_processor_->schedule(sedp_.max_type_lookup_service_reply_period_);`
This may not be needed, as this part of code does not create additional entries in the `MatchingDataMap` to be removed if the reply times out.